agent/host.go

```diff
@@ -134,5 +135,52 @@ func createRootNode(g *graph.Graph) (*graph.Node, error) {
 		m.SetField("VirtualizationRole", hostInfo.VirtualizationRole)
 	}
 
+	dmi := dmidecode.New()
+	if err := dmi.Run(); err == nil {
+		// If we have a valid DMI table
+		dmi_data := map[string]interface{}{}
+
+		// Let's scan every DMI type
+		for type_number := 0; type_number < 128; type_number++ {
+			// For each DMI type
+			byTypeData, byTypeErr := dmi.SearchByType(type_number)
```

Comment:
rather reuse `err` as we process each error immediately no need to have a per operation error variable.